diff --git a/fast_adaptive_boundary.py b/fast_adaptive_boundary.py
index 54a77f2..0a2d853 100644
--- a/fast_adaptive_boundary.py
+++ b/fast_adaptive_boundary.py
@@ -11,9 +11,17 @@ from __future__ import print_function
 from __future__ import unicode_literals

 import torch
-from torch.autograd.gradcheck import zero_gradients
 import time

+def zero_gradients(x):
+    if isinstance(x, torch.Tensor):
+        if x.grad is not None:
+            x.grad.detach_()
+            x.grad.zero_()
+    elif isinstance(x, collections.abc.Iterable):
+        for elem in x:
+            zero_gradients(elem)
+
 try:
     from torch import flip
 except ImportError:
